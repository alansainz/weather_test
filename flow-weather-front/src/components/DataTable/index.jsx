import React from 'react';

import { 
  Container,
  Title,
  Data,
  DataContainer,
  Header
} from './style.js'

const metersToKm = (meters) => meters / 1000;

const DataTable = ({ windSpeed, humidity, pressure, visibility }) => (
  <Container>
    <DataContainer>

      <Header>Extra Info</Header>

      <Title>Wind</Title>
      <Data>{windSpeed && `${windSpeed} km/h`} </Data>
      
      <Title>Humidity</Title>
      <Data>{humidity && `${humidity} %`} </Data>
      
      <Title>Pressure</Title>
      <Data>{pressure && `${pressure} mb`} </Data>
      
      <Title>Visibility</Title>
      <Data isLast={true}>{visibility && `${metersToKm(visibility)} km`}</Data>
     
    </DataContainer>
  </Container>
);

export default DataTable