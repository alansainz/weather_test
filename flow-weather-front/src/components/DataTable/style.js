import styled from 'styled-components';

import { TABLE_COLOR } from '../../config/styleConstants.js';

export const Container = styled.div`
  width: 150px;
  height: 350px;
`

export const Title = styled.div`
  font-weight: bold;
  font-size: 16px;
`
export const Header = styled.div`
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
`

export const Data = styled.div`
  font-size: 16px;
  padding-bottom: 5px;
  margin-bottom: 5px;
  color: ${TABLE_COLOR}
  border-bottom: ${props => !props.isLast ? `1px solid ${TABLE_COLOR}` : null};
`

export const DataContainer = styled.div`
  padding: 20px
`


