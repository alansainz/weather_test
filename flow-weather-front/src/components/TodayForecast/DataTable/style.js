import styled from 'styled-components';

export const Container = styled.div`
  width: 150px;
  height: 350px;
`;
export const Header = styled.div`
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
`;

export const DataContainer = styled.div`
  padding: 20px;
`;
