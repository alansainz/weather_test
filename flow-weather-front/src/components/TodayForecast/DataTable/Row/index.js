import React from 'react';

import { Title, Data } from './style.js';

const Row = ({ title, value, symbol, isLast }) => (
  <React.Fragment>
    <Title>{title}</Title>
    <Data isLast={isLast}>{value && `${value} ${symbol}`} </Data>
  </React.Fragment>
);

export default Row;
