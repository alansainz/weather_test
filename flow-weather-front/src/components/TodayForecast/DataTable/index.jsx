import React from 'react';
import Row from './Row'
import { 
  Container,
  DataContainer,
  Header
} from './style.js'

const metersToKm = (meters) => meters / 1000;

const DataTable = ({ windSpeed, humidity, pressure, visibility }) => {
  const tableData = [
    {
      title: 'Wind',
      symbol: 'km/h',
      value: windSpeed && windSpeed
    },
    {
      title: 'Humidity',
      symbol: '%',
      value: humidity && humidity
    },
    {
      title: 'Pressure',
      symbol: 'mb',
      value: pressure && pressure
    },
    {
      title: 'Visibility',
      symbol: 'km',
      value: visibility && metersToKm(visibility),
      isLast: true
    }
  ]
  return (
    <Container>
      <DataContainer>
        <Header>Extra Info</Header>
        {
          tableData.map((row, index) => 
            <Row 
              key={index}
              title={row.title} 
              value={row.value} 
              symbol={row.symbol}
            />
          )
        }
      </DataContainer>
    </Container>
  )
};

export default DataTable