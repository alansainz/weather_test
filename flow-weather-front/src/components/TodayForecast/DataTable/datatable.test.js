import React from 'react';
import ReactDOM from 'react-dom';
import DataTable from './index.jsx';
import { shallow  } from 'enzyme';

describe("DataTable Component", () => {
  it('should render with no props', () => {
    const component = shallow(<DataTable />);
    expect(component).toMatchSnapshot();
  });

  it('should render with given props', () => { 
    const exampleData = {
      windSpeed: 23, 
      humidity: 1100, 
      pressure: 23, 
      visibility: 2000
    }
    const component = shallow(<DataTable {...exampleData} />);
    expect(component).toMatchSnapshot();
  });
})