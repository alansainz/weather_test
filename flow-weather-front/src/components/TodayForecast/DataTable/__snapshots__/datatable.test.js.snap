// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Component should render with given props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataTable
    humidity={1100}
    pressure={23}
    visibility={2000}
    windSpeed={23}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          Extra Info
        </ForwardRef>
        <Row
          symbol="km/h"
          title="Wind"
          value={23}
        />
        <Row
          symbol="%"
          title="Humidity"
          value={1100}
        />
        <Row
          symbol="mb"
          title="Pressure"
          value={23}
        />
        <Row
          symbol="km"
          title="Visibility"
          value={2}
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Extra Info
          </ForwardRef>,
          Array [
            <Row
              symbol="km/h"
              title="Wind"
              value={23}
            />,
            <Row
              symbol="%"
              title="Humidity"
              value={1100}
            />,
            <Row
              symbol="mb"
              title="Pressure"
              value={23}
            />,
            <Row
              symbol="km"
              title="Visibility"
              value={2}
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Extra Info",
          },
          "ref": null,
          "rendered": "Extra Info",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "symbol": "km/h",
            "title": "Wind",
            "value": 23,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "symbol": "%",
            "title": "Humidity",
            "value": 1100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "symbol": "mb",
            "title": "Pressure",
            "value": 23,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "symbol": "km",
            "title": "Visibility",
            "value": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
  padding: 20px
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  width: 150px;
  height: 350px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            Extra Info
          </ForwardRef>
          <Row
            symbol="km/h"
            title="Wind"
            value={23}
          />
          <Row
            symbol="%"
            title="Humidity"
            value={1100}
          />
          <Row
            symbol="mb"
            title="Pressure"
            value={23}
          />
          <Row
            symbol="km"
            title="Visibility"
            value={2}
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Extra Info
            </ForwardRef>,
            Array [
              <Row
                symbol="km/h"
                title="Wind"
                value={23}
              />,
              <Row
                symbol="%"
                title="Humidity"
                value={1100}
              />,
              <Row
                symbol="mb"
                title="Pressure"
                value={23}
              />,
              <Row
                symbol="km"
                title="Visibility"
                value={2}
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Extra Info",
            },
            "ref": null,
            "rendered": "Extra Info",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "symbol": "km/h",
              "title": "Wind",
              "value": 23,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "symbol": "%",
              "title": "Humidity",
              "value": 1100,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "symbol": "mb",
              "title": "Pressure",
              "value": 23,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "symbol": "km",
              "title": "Visibility",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
  padding: 20px
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  width: 150px;
  height: 350px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`DataTable Component should render with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          Extra Info
        </ForwardRef>
        <Row
          symbol="km/h"
          title="Wind"
        />
        <Row
          symbol="%"
          title="Humidity"
        />
        <Row
          symbol="mb"
          title="Pressure"
        />
        <Row
          symbol="km"
          title="Visibility"
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Extra Info
          </ForwardRef>,
          Array [
            <Row
              symbol="km/h"
              title="Wind"
            />,
            <Row
              symbol="%"
              title="Humidity"
            />,
            <Row
              symbol="mb"
              title="Pressure"
            />,
            <Row
              symbol="km"
              title="Visibility"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Extra Info",
          },
          "ref": null,
          "rendered": "Extra Info",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "symbol": "km/h",
            "title": "Wind",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "symbol": "%",
            "title": "Humidity",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "symbol": "mb",
            "title": "Pressure",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "symbol": "km",
            "title": "Visibility",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
  padding: 20px
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  width: 150px;
  height: 350px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            Extra Info
          </ForwardRef>
          <Row
            symbol="km/h"
            title="Wind"
          />
          <Row
            symbol="%"
            title="Humidity"
          />
          <Row
            symbol="mb"
            title="Pressure"
          />
          <Row
            symbol="km"
            title="Visibility"
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Extra Info
            </ForwardRef>,
            Array [
              <Row
                symbol="km/h"
                title="Wind"
              />,
              <Row
                symbol="%"
                title="Humidity"
              />,
              <Row
                symbol="mb"
                title="Pressure"
              />,
              <Row
                symbol="km"
                title="Visibility"
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Extra Info",
            },
            "ref": null,
            "rendered": "Extra Info",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
  font-weight: bold;
  font-size: 20px;
  margin-bottom: 10px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "symbol": "km/h",
              "title": "Wind",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "symbol": "%",
              "title": "Humidity",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "symbol": "mb",
              "title": "Pressure",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "symbol": "km",
              "title": "Visibility",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
  padding: 20px
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  width: 150px;
  height: 350px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
